base_layer/wallet_ffi/.venv/lib/python3.12/site-packages/_pytest/config/__init__.py

```
@@ -0,0 +1,2029 @@
+        if opts is None:
+            method = getattr(module_or_class, name)
+            if name.startswith("pytest_"):
+                legacy = _get_legacy_hook_marks(
+                    method, "spec", ("firstresult", "historic")
+                )
+                opts = cast(HookspecOpts, legacy)
+        return opts
+
+    def register(self, plugin: _PluggyPlugin, name: str | None = None) -> str | None:
+        if name in _pytest.deprecated.DEPRECATED_EXTERNAL_PLUGINS:
+            warnings.warn(
```

Ruff (0.11.9)

No explicit `stacklevel` keyword argument found

Set `stacklevel=2`

(B028)